src/js/ripple/amount.js

```diff
@@ -798,29 +576,41 @@ Amount.prototype.parse_quality = function(quality, counterCurrency, counterIssue
 
   var baseCurrency = Currency.from_json(opts.base_currency);
 
-  this._is_negative = false;
-  this._value       = new BigInteger(quality.substring(quality.length-14), 16);
-  this._offset      = parseInt(quality.substring(quality.length-16, quality.length-14), 16)-100;
+  var mantissa_hex  = quality.substring(quality.length-14);
+  var offset_hex    = quality.substring(quality.length-16, quality.length-14);
+  var mantissa      = new BigNumber(mantissa_hex, 16);
+  var offset        = parseInt(offset_hex, 16) - 100;
+
+  var value = new BigNumber(mantissa.toString() + 'e' + offset.toString());
+
   this._currency    = Currency.from_json(counterCurrency);
   this._issuer      = UInt160.from_json(counterIssuer);
   this._is_native   = this._currency.is_native();
 
+  var exponent = 0;
+  if (this._is_native) {
+    if (opts.inverse) {
+      exponent += 1;
+    } else {
+      exponent -= 1;
```

Comment:
I don't really get it. What case will the power be -2 or 2 ?
We don't have any tests for the cases where the power is -2, 2 it seems.